scriptler/addShellPreBuildStepToProject.groovy

```diff
@@ -0,0 +1,42 @@
+/*** BEGIN META {
+ "name" : "Add a pre-build shell script step",
+ "comment" : "Add a pre-build shell script step to all Maven and Frestyle Projects",
+ "parameters" : ["filePath"],
+ "core": "1.609",
+ "authors" : [
+ { name : "Kuisathaverat" }
+ ]
+ } END META**/
+
+import hudson.util.*
+import hudson.tasks.*
+import hudson.maven.*
+
+//def filePath = '/path/to/script.sh'
+def scriptFile = new File(filePath)
+
+if(!scriptFile.exists() && scriptFile.isDirectory()) { 
+  throw IOException("Check file name")
+}
+String script = scriptFile.text
+
+Jenkins.instance.getAllItems(Job.class)
+	.findAll{ it instanceof FreeStyleProject || it instanceof MavenModuleSet  }
+	.each{
+      println it.name + " - " + it.class
+
+      DescribableList<Builder,Descriptor<Builder>> builders = new DescribableList<Builder,Descriptor<Builder>>()
```

Comment:
Im surprised this works as the constructor is `protected` scope. 

```
    protected DescribableList() {
    }
```

But assuming it does so just a comment.